tsl/src/continuous_aggs/create.c

```diff
@@ -2238,20 +2358,39 @@ finalizequery_init(FinalizeQueryInfo *inp, Query *orig_query, MatTableColumnInfo
  * for the materialization table
  * matcollist - column list for mat table
  * mattbladdress - materialization table ObjectAddress
+ * This is the function responsible for creating the final
+ * structures for selecting from the materialized hypertable
+ * created for the Cagg which is
+ * select * from _timescaldeb_internal._matrialized_hypertable_<xxx>
  */
 static Query *
 finalizequery_get_select_query(FinalizeQueryInfo *inp, List *matcollist,
-							   ObjectAddress *mattbladdress)
+							   ObjectAddress *mattbladdress, char *relname)
 {
 	Query *final_selquery = NULL;
 	ListCell *lc;
+	FromExpr *fromexpr;
+	RangeTblEntry *rte;
+
 	/*
 	 * for initial cagg creation rtable will have only 1 entry,
 	 * for alter table rtable will have multiple entries with our
 	 * RangeTblEntry as last member.
+	 * For cagg with joins, we need to create a new RTE and jointree
+	 * which contains the information of the materialised hypertable
+	 * that is created for this cagg.
 	 */
-	RangeTblEntry *rte = llast_node(RangeTblEntry, inp->final_userquery->rtable);
-	FromExpr *fromexpr;
+	if (list_length(inp->final_userquery->jointree->fromlist) > 1)
```

Comment:
```suggestion
	if (list_length(inp->final_userquery->jointree->fromlist) >= CONTINUOUS_AGG_MAX_JOIN_RELATIONS)
```
Shouldn't be this check in that way???